### ISSUE TITLE
Are Rasterizer outputs differentiable?

### ISSUE THREAD

**Issue by YuxuanSnow** [facebookresearch/pytorch3d#905]
Dear developers,

thank you for the amazing work!

I wonder if the rasterizer outputs such as `pix_to_face`, `dists`, `bary_coords`, or `zbuf` are differentiable over given mesh? I tried to print `.requires_grad`, the rendered image is true, but the rasterization results are all false. 

Here are my code:
```
renderer = MeshRendererWithFragments(
            rasterizer=MeshRasterizer(
                cameras=self.cameras,
                raster_settings=raster_settings
            ),
            shader=SoftPhongShader(
                device=device, 
                lights=lights, 
                blend_params=blend_params,
                cameras=self.cameras
            )
        )

images, frag = self.renderer(mesh, lights=lights)
            
pix_to_face = frag.pix_to_face

print(images.requires_grad)                  # result is true
print(pix_to_face.requires_grad)           # result is false
```

Is it possible to turn on gradient on pix_to_face and make it related to mesh? Then i can do backward propogation over pix_to_face and optimize over mesh.
